server/src/main/java/org/opensearch/rest/action/cat/RestCatSegmentReplicationAction.java

```diff
@@ -0,0 +1,189 @@
+import org.opensearch.action.support.IndicesOptions;
+import org.opensearch.client.node.NodeClient;
+import org.opensearch.common.Strings;
+import org.opensearch.common.Table;
+import org.opensearch.common.unit.TimeValue;
+import org.opensearch.common.xcontent.XContentOpenSearchExtension;
+import org.opensearch.indices.replication.SegmentReplicationState;
+import org.opensearch.rest.BaseRestHandler;
+import org.opensearch.rest.RestHandler;
+import org.opensearch.rest.RestRequest;
+import org.opensearch.rest.RestResponse;
+import org.opensearch.rest.action.RestResponseListener;
+
+import java.util.Comparator;
+import java.util.List;
+import java.util.Locale;
+
+import static java.util.Arrays.asList;
+import static java.util.Collections.unmodifiableList;
+import static org.opensearch.rest.RestRequest.Method.GET;
+
+public class RestCatSegmentReplicationAction extends AbstractCatAction {
+    @Override
+    public List<RestHandler.Route> routes() {
+        return unmodifiableList(
+            asList(new RestHandler.Route(GET, "/_cat/segment_replication"), new RestHandler.Route(GET, "/_cat/segment_replication/{index}"))
+        );
+    }
+
+    @Override
+    public String getName() {
+        return "cat_segment_replication_action";
+    }
+
+    @Override
+    protected void documentation(StringBuilder sb) {
+        sb.append("/_cat/segment_replication\n");
+        sb.append("/_cat/segment_replication/{index}\n");
+    }
+
+    @Override
+    public BaseRestHandler.RestChannelConsumer doCatRequest(final RestRequest request, final NodeClient client) {
+        final SegmentReplicationRequest segmentReplicationRequest = new SegmentReplicationRequest(
+            Strings.splitStringByCommaToArray(request.param("index"))
+        );
+        segmentReplicationRequest.timeout(request.param("timeout"));
+        segmentReplicationRequest.detailed(request.paramAsBoolean("detailed", false));
+        segmentReplicationRequest.activeOnly(request.paramAsBoolean("active_only", false));
+        segmentReplicationRequest.indicesOptions(IndicesOptions.fromRequest(request, segmentReplicationRequest.indicesOptions()));
+
+        return channel -> client.admin()
+            .indices()
+            .segment_replication(segmentReplicationRequest, new RestResponseListener<SegmentReplicationResponse>(channel) {
+                @Override
+                public RestResponse buildResponse(final SegmentReplicationResponse response) throws Exception {
+                    return RestTable.buildResponse(buildSegmentReplicationTable(request, response), channel);
+                }
+            });
+    }
+
+    @Override
+    protected Table getTableWithHeader(RestRequest request) {
+
+        Table t = new Table();
+        t.startHeaders()
+            .addCell("index", "alias:i,idx;desc:index name")
+            .addCell("shardId", "desc: shard Id")
+            .addCell("replication_id", "desc: replication Id")
```

Comment:
What is the logic for calculating this replication_id? Is it unique per segment level, per-index level, etc?

Reply:
Whenever a new segrep event starts on a replica shard, we assign an id for that specific segrep event to track. It is [auto incremented](https://github.com/opensearch-project/OpenSearch/blob/f9eb9bfe17b71aae7e2a00ac0337909f1b3e61e7/server/src/main/java/org/opensearch/indices/replication/common/ReplicationTarget.java#L86) on every new segrep event. This is unique per shard level.

Reply:
Got it, just to be clear maybe we can mention that it's unique per shard level somewhere in the docs/comments.